pkg/plugin/validator.go

```diff
@@ -62,26 +62,26 @@ func NewValidator(ghClinet *github.Client, ghApp *githubapp.GitHubApp) *Validato
 }
 
 // MatchIssue parses issue info from provided issueURL and validate if the issue is valid.
-func (v *Validator) MatchIssue(ctx context.Context, issueURL string) error {
+func (v *Validator) MatchIssue(ctx context.Context, issueURL string) (*pluginGitHubIssue, error) {
 	info, err := parseIssueInfoFromURL(issueURL)
 	if err != nil {
-		return errors.Join(errInvalidJustification, fmt.Errorf("failed to parse issueURL: %w", err))
+		return nil, errors.Join(errInvalidJustification, fmt.Errorf("failed to parse issueURL: %w", err))
```

Comment:
I think we should prefer the double %w format instead:

```suggestion
		return nil, fmt.Errorf("%w: failed to parse issueURL: %w", errInvalidJustification, err)
```

That will work on Go 1.20+.

Reply:
Thanks! I also updated other places where I previously used `errors.Join`

Reply:
`errors.Join` is useful when one of the errors could be nil, for example:

```go
result, err := doSomething()
return errors.Join(err, fmt.Errorf("oops")
```